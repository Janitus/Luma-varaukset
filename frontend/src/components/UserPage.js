import React from 'react'
import { useTranslation } from 'react-i18next'
import { useHistory } from 'react-router'

const UserPage = ({ currentUser, setShowEventForm }) => {
  const { t } = useTranslation('user')
  const history = useHistory()

  const createEvent = (event) => {
    event.preventDefault()
    setShowEventForm(true)
  }

  const listUsers = (event) => {
    event.preventDefault()
    history.push('/users')
  }

  const listVisits = (event) => {
    event.preventDefault()
    history.push('/visits')
  }

  const extras = (event) => {
    event.preventDefault()
    history.push('/extras')
  }

  const forms = (event) => {
    event.preventDefault()
    history.push('/forms')
  }

  const eventList = (event) => {
    event.preventDefault()
    history.push('/events')
  }

  const emailConfig = (event) => {
    event.preventDefault()
    history.push('/email-config')
  }

  if (!currentUser) return <div></div>

  return (
    <div className="field is-grouped">
      <p className="control">
        <button className="button luma" onClick={createEvent}>{t('create-visit')}</button>
      </p>
      <p className="control">
        <button className="button luma" onClick={listVisits}>{t('reservations')}</button>
      </p>
      <p className="control">
        <button className="button luma" onClick={extras}>{t('extras')}</button>
      </p>
      <p className="control">
        <button className="button luma" onClick={forms}>{t('forms')}</button>
      </p>
      {currentUser.isAdmin &&
        <>
          <p className="control">
            <button className="button luma" onClick={listUsers}>{t('user-list')}</button>
          </p>
          <p className="control">
            <button className="button luma" onClick={eventList}>{t('event-list')}</button>
          </p>
          <p className="control">
            <button className="button luma" onClick={emailConfig}>{t('email-config')}</button>
          </p>
        </>
      }
    </div>
  )
}

export default UserPage