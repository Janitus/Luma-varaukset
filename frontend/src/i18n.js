import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import fi from './locales/fi/translations.json'
import en from './locales/en/translations.json'
import sv from './locales/sv/translations.json'

export const resources = {
  fi: {
    translation: fi,
    language: 'fi'
  },
  en: {
    translation: en,
    language: 'en'
  },
  sv: {
    translation: sv,
    language: 'sv'
  }
}

i18n
  .use(initReactI18next)
  .init({
    resources,
    fallbackLng: 'fi',
    debug: false,
    interpolation: {
      escapeValue: false,
    }
  })

export const { t } = i18n

export default i18n
