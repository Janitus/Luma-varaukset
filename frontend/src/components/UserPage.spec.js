import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render, waitFor, fireEvent } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'
import UserPage from './UserPage'
import '../i18n'

test('Renders without error with empty user', async () => {
  const { container } = render(
    <UserPage/>
  )
  await waitFor(() => new Promise((res) => setTimeout(res, 0))) // Allow component time to render
  expect(container).toMatchSnapshot()
})

test('Renders without error with user', async () => {
  const { container } = render(
    <UserPage  currentUser={{ name: 'Tester', isAdmin: false }}/>
  )
  await waitFor(() => new Promise((res) => setTimeout(res, 0))) // Allow component time to render
  expect(container).toMatchSnapshot()
})

test('Renders without error with admin user', async () => {
  const { container } = render(
    <UserPage  currentUser={{ name: 'Tester', isAdmin: true }}/>
  )
  await waitFor(() => new Promise((res) => setTimeout(res, 0))) // Allow component time to render
  expect(container).toMatchSnapshot()
})

test('CreateEvent modal is set to true', async () => {
  const handler = jest.fn()

  const { container } = render(
    <UserPage currentUser={{ name: 'Tester', isAdmin: false }} setShowEventForm={handler}/>
  )
  await waitFor(() => new Promise((res) => setTimeout(res, 0))) // Allow component time to render
  const buttons = container.querySelectorAll('.button.luma')
  fireEvent.click(buttons[0])
  expect(handler).toHaveBeenCalledTimes(1)
  expect(handler).toHaveBeenCalledWith(true)
})

test('History should be set to /visits', async () => {
  const history = createMemoryHistory()

  const { container } = render(
    <Router history={history}>
      <UserPage currentUser={{ name: 'Tester', isAdmin: false }}/>
    </Router>
  )
  await waitFor(() => new Promise((res) => setTimeout(res, 0))) // Allow component time to render
  const buttons = container.querySelectorAll('.button.luma')
  fireEvent.click(buttons[1])
  expect(history.location.pathname).toBe('/visits')
})

test('History should be set to /extras', async () => {
  const history = createMemoryHistory()

  const { container } = render(
    <Router history={history}>
      <UserPage currentUser={{ name: 'Tester', isAdmin: false }}/>
    </Router>
  )
  await waitFor(() => new Promise((res) => setTimeout(res, 0))) // Allow component time to render
  const buttons = container.querySelectorAll('.button.luma')
  fireEvent.click(buttons[2])
  expect(history.location.pathname).toBe('/extras')
})

test('History should be set to /forms', async () => {
  const history = createMemoryHistory()

  const { container } = render(
    <Router history={history}>
      <UserPage currentUser={{ name: 'Tester', isAdmin: true }}/>
    </Router>
  )
  await waitFor(() => new Promise((res) => setTimeout(res, 0))) // Allow component time to render
  const buttons = container.querySelectorAll('.button.luma')
  fireEvent.click(buttons[3])
  expect(history.location.pathname).toBe('/forms')
})

test('History should be set to /users', async () => {
  const history = createMemoryHistory()

  const { container } = render(
    <Router history={history}>
      <UserPage currentUser={{ name: 'Tester', isAdmin: true }}/>
    </Router>
  )
  await waitFor(() => new Promise((res) => setTimeout(res, 0))) // Allow component time to render
  const buttons = container.querySelectorAll('.button.luma')
  fireEvent.click(buttons[6])
  expect(history.location.pathname).toBe('/users')
})