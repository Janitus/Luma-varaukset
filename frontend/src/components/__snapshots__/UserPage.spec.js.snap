// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without error with admin user 1`] = `
<div>
  <div
    class="field is-grouped"
  >
    <p
      class="control"
    >
      <button
        class="button luma"
      >
        Luo uusi vierailu
      </button>
    </p>
    <p
      class="control"
    >
      <button
        class="button luma"
      >
        Varaukset
      </button>
    </p>
    <p
      class="control"
    >
      <button
        class="button luma"
      >
        Lisäpalvelut
      </button>
    </p>
    <p
      class="control"
    >
      <button
        class="button luma"
      >
        Lomakkeet
      </button>
    </p>
    <p
      class="control"
    >
      <button
        class="button luma"
      >
        Käyttäjälista
      </button>
    </p>
    <p
      class="control"
    >
      <button
        class="button luma"
      >
        Vierailulista
      </button>
    </p>
    <p
      class="control"
    >
      <button
        class="button luma"
      >
        Sähköposti konfiguraatiot
      </button>
    </p>
  </div>
</div>
`;

exports[`Renders without error with empty user 1`] = `
<div>
  <div />
</div>
`;

exports[`Renders without error with user 1`] = `
<div>
  <div
    class="field is-grouped"
  >
    <p
      class="control"
    >
      <button
        class="button luma"
      >
        Luo uusi vierailu
      </button>
    </p>
    <p
      class="control"
    >
      <button
        class="button luma"
      >
        Varaukset
      </button>
    </p>
    <p
      class="control"
    >
      <button
        class="button luma"
      >
        Lisäpalvelut
      </button>
    </p>
    <p
      class="control"
    >
      <button
        class="button luma"
      >
        Lomakkeet
      </button>
    </p>
  </div>
</div>
`;
